src/capi/azext_capi/custom.py

```diff
@@ -520,17 +521,12 @@ def create_workload_cluster(  # pylint: disable=too-many-arguments,too-many-loca
         "PRE_BOOTSTRAP_CMDS": bootstrap_cmds["pre"],
         "POST_BOOTSTRAP_CMDS": bootstrap_cmds["post"],
         "AKS_INFRA_RG_NAME": os.environ.get(AKS_INFRA_RG_NAME, None),
-        "AKS_VNET_NAME": os.environ.get(AKS_VNET_NAME, None)
+        "AKS_VNET_NAME": os.environ.get(AKS_VNET_NAME, None),
+        "EXTERNAL_CLOUD_PROVIDER": external_cloud_provider,
```

Comment:
This change is paired with the previous one that doesn't prevent the user from adding these flags/variables if they know what they're doing.